tests/api/test_cluster.cpp

```diff
@@ -0,0 +1,76 @@
+
+#include <gtest/gtest.h>
+#include "fmt/xchar.h"
+
+#include <algorithm>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include "tests/test_utils/generate_cluster_desc.hpp"
+
+// TODO: change to tt_cluster
+#include "device/tt_device.h"
+#include "device/tt_cluster_descriptor.h"
+
+// TODO: do proper renaming.
+using Cluster = tt_SiliconDevice;
+
+// These tests are intended to be run with the same code on all kinds of systems:
+// E75, E150, E300
+// N150. N300
+// Galaxy
+
+// This test should be one line only.
+TEST(ApiTest, OpenAllChips) {
+
+    // TODO: This should not be needed. And could be part of the cluster descriptor probably.
+    // Note that cluster descriptor holds logical ids of chips.
+    // Which are different than physical PCI ids, which are /dev/tenstorrent/N ones.
+    // You have to see if physical PCIe is GS before constructing a cluster descriptor.
+    std::vector<int> pci_device_ids = PCIDevice::enumerate_devices();
+    std::set<int> pci_device_ids_set;
+    for (int pci_device_id : pci_device_ids) {
+        pci_device_ids_set.insert(pci_device_id);
+    }
+
+    // TODO: This should be removed from the API, the driver itself should do it.
+    std::string soc_path;
+    int physical_device_id = pci_device_ids[0];
```

Comment:
This is undefined if `pci_device_ids` is empty, maybe an early exit + a warning ("no chips detected") in that case?

Reply:
Left another TODO in test to make sure Cluster() doesn't throw on a system without TT device.